tests/integration/tests/live.test.ts

```diff
@@ -0,0 +1,164 @@
+import { describe, expect, test } from "bun:test";
+import { type LiveAction, RecordId, ResponseError, Uuid } from "../../../src";
+import { setupServer } from "../surreal.ts";
+
+const { createSurreal } = await setupServer();
+
+async function withTimeout(p: Promise<void>, ms: number): Promise<void> {
+	const { promise, resolve, reject } = Promise.withResolvers<void>();
+	const timeout = setTimeout(() => reject(new Error("Timeout")), ms);
+
+	await Promise.race([
+		promise,
+		p.then(resolve).finally(() => clearTimeout(timeout)),
+	]);
+}
+
+describe("Live Queries", async () => {
+	const surreal = await createSurreal();
+
+	test("live", async () => {
+		if (surreal.connection?.connection.url?.protocol.startsWith("http")) {
+			expect(surreal.live("person")).rejects.toBeInstanceOf(ResponseError);
+		} else {
+			const events: {
+				action: LiveAction;
+				result: Record<string, unknown>;
+			}[] = [];
+			const { promise, resolve } = Promise.withResolvers<void>();
+
+			const queryUuid = await surreal.live("person", (action, result) => {
+				events.push({ action, result });
+				if (action === "DELETE") resolve();
+			});
+
+			expect(queryUuid).toBeInstanceOf(Uuid);
+
+			await surreal.create(new RecordId("person", 1), {
+				firstname: "John",
+				lastname: "Doe",
+			});
+			await surreal.update(new RecordId("person", 1), {
+				firstname: "Jane",
+				lastname: "Doe",
+			});
+			await surreal.delete(new RecordId("person", 1));
+
+			await withTimeout(promise, 5e3); // Wait for the DELETE event
```

Comment:
This can be flaky, I would have a helper function to wait for event with timeout if failed to get